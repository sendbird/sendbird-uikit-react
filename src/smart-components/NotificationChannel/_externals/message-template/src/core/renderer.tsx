import React from 'react';

import type { BasicProps, ComponentType, ComponentsUnion, GetProperties } from '../types/components';

type FC<Props, ParsedStyle> = (props: BasicProps<Props, ParsedStyle>) => React.ReactElement | null;

type ComponentProperties<Type extends ComponentType> = GetProperties<Type, ComponentsUnion>;

interface RendererOptions<ParsedStyle> {
  views?: {
    [component in ComponentType]?: FC<ComponentProperties<component>, ParsedStyle>;
  };
}

export type Renderer<ParsedStyle> = {
  [component in ComponentType]: FC<ComponentProperties<component>, ParsedStyle>;
};

const FRAGMENT = ({ children }: React.PropsWithChildren) => <React.Fragment>{children}</React.Fragment>;

export function createRenderer<ParsedStyle = object | string>(
  opts?: RendererOptions<ParsedStyle>,
): Renderer<ParsedStyle> {
  return {
    box: opts?.views?.box || FRAGMENT,
    text: opts?.views?.text || FRAGMENT,
    image: opts?.views?.image || FRAGMENT,
    imageButton: opts?.views?.imageButton || FRAGMENT,
    textButton: opts?.views?.textButton || FRAGMENT,
  };
}
