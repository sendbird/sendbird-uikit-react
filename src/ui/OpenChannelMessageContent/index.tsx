import React, { ReactElement } from 'react';
import './index.scss';

interface Props {
}

export default function OpenChannelMessageContent({}: Props): ReactElement {
  return (
    <div>
    </div>
  );
}
