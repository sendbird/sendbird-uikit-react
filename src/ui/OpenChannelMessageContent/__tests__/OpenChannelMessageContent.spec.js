import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import OpenChannelMessageContent from "../index";

describe('OpenChannelMessageContent', () => {
  it('should do a snapshot test of the OpenChannelMessageContent DOM', function() {
    const text = "example-text";
    const component = renderer.create(
      <OpenChannelMessageContent />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
