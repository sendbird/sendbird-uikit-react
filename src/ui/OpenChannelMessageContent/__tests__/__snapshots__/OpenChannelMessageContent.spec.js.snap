// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenChannelMessageContent should do a snapshot test of the OpenChannelMessageContent DOM 1`] = `<div />`;
